contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -936,22 +936,21 @@ Datum getutcdate(PG_FUNCTION_ARGS)
 
 Datum getdate_internal(PG_FUNCTION_ARGS)
 {
-	PG_RETURN_TIMESTAMP(DirectFunctionCall2(timestamp_trunc,CStringGetTextDatum("millisecond"),
-						PointerGetDatum(GetCurrentStatementStartTimestamp())));
+	PG_RETURN_DATUM(DirectFunctionCall1(common_utility_plugin_ptr->timestamptz_datetime, 
+						DirectFunctionCall2(timestamp_trunc,CStringGetTextDatum("millisecond"),
+											TimestampTzGetDatum(GetCurrentStatementStartTimestamp()))));
```

Comment:
`timestamp_trunc` returns `timestamp` whereas `timestamptz_datetime` is expecting `TimestampTz`, how is both compliant and working?

Reply:
The compatibility between timestamp_trunc and timestamptz_datetime is achieved through implicit casting in PostgreSQL.

This implicit casting behavior is defined in the PostgreSQL source code and is part of the built-in type conversion rules. It allows for seamless interoperability between timestamp and TimestampTz values in function calls and expressions.
The implicit cast from timestamp to TimestampTz preserves the original time value but adds or removes the time zone offset based on the current time zone setting. This ensures that the timestamptz_datetime function receives a valid TimestampTz value, even though the timestamp_trunc function returns a timestamp value.

But yes, we shouldn't rely on implicit casting, Added Explicit Cast to achieve this.